### DISCUSSION TITLE
Bridge deposit

### DISCUSSION THREAD

**Discussion by Winless** [zkSync-Community-Hub/zksync-developers#106] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

using bridge to deposit

### What is the specific issue or error you're encountering?

 It is currently not possible to deposit new ERC20 tokens into the zksync era testnet through L1bridge, and all transactions are failing. What could be the reason for this?

### Can you share the error messages or logs you're receiving, if any?

https://zksync2-testnet.zkscan.io/tx/0xd2891f4bfbc762efa36c90d043800376697539e0935edfe8ba99d6b540469830/internal-transactions

the tx shows

### Have you made any recent changes to the contract before encountering this issue?

Regardless of whether I use the official L1bridge or my own custom L1 contract to call the bridge, all attempts have failed.

### Are there any external libraries or contracts that your contract interacts with?

no

### Can you provide the relevant portions of your contract code where the issue is occurring?

no

### Have you tried to isolate the problem, and if so, what were the results?

always failed in l2 and shows Error: (Awaiting internal transactions for reason)

### What steps have you already taken to try to resolve the issue?

I have previously successfully deposited, but I am unable to deposit using the new ERC20 token. However, if I use a previously deployed ERC20 address, the deposit is successful. I believe the issue lies with deploying the new token on L2.

**Comment by uF4No** [comment]
hey @Winless 

Can you confirm if you're getting this error using the portal UI or a script to interact directly with the bridge contract? 
If it's a script, can you share the code?

Thanks

**Comment by Romsters** [comment, marked as answer]
Hi @Winless, did you approve your new ERC20 token before bridging it? https://era.zksync.io/docs/api/js/accounts-l1-l2.html#approving-deposit-of-tokens

**Reply by MexicanAce** [reply]
Closing this discussion due to lack of response. Marking the response above as the answer as it links to more information in the docs.

**Comment by casesumteryan** [comment]
It seems there may be an issue with deploying the new ERC20 token on L2, resulting in failed deposit transactions. Further investigation into the token deployment process on L2 may be necessary to resolve this issue.

**Comment by atzipajabken** [comment]
The user is encountering an issue with depositing new ERC20 tokens into the zksync era testnet through L1bridge, with all transactions failing and showing an "Awaiting internal transactions for reason" error. The user has not made any recent changes to their contract and is not interacting with any external libraries or contracts. The user has tried using both the official L1bridge and their own custom L1 contract but has encountered the same issue. The user suspects the issue may be related to deploying the new token on L2.
